### DISCUSSION TITLE
watch and listen to the same provider

### DISCUSSION THREAD

**Discussion by aidanmack** [rrousselGit/riverpod#3779] [Q&A]
So I listen to the provider for things like navigation and show popup and use watch for changes to a widget state.

But whats best practice for the watch to not react to a listen state?

So if I emit down `showPopup` the watch still need to react to this when in-fact there is no state change and it should carry on showing its current state (widget)

'showpopup' is strictly for the listen to react to.

**Comment by gdurandrexel** [comment]
Don't use the same provider, create 2 providers for 2 different states since, clearly, they do not represent the same thing.

Alternatively you could use `select` to watch only specific parts of your state, but that's probably not the best solution.

**Comment by rrousselGit** [comment]
You're free to use both watch and listen on the same provider.
There's no restriction around that.

**Comment by gdurandrexel** [comment]
My understanding of the problem here is that the same provider provides a state containing values that track popup notifications and others that are used to build a widget content: if the state is updated to "notify" a popup notification then the widget is also rebuilt, which is not wanted.

**Reply by aidanmack** [reply]
Exactly @gdurandrexel, as you said that would be my solution here. 2 providers - one for state and the other for "events" like navigation, show bottom sheet ect.

I just wanted check if theirs away to do both only using one provider.
